src/web/src/app.tsx

```diff
@@ -0,0 +1,362 @@
+  filter.minPriority =
+    currentSeed === SeedType.UPSTREAM
+      ? StudyPriority.STABLE_MIN
+      : StudyPriority.NON_INTERESTING;
+  try {
+    const priorityString = searchParams.get('minPriority');
+    if (priorityString != null) {
+      filter.minPriority = parseInt(priorityString);
+    }
+  } catch {
+    /* empty */
+  }
+  filter.showEmptyGroups = searchParams.get('showEmptyGroups') === 'true';
+  const toggleShowEmptyGroups = (): void => {
+    setParam('showEmptyGroups', filter.showEmptyGroups ? 'false' : 'true');
+  };
+
+  filter.includeOutdated = searchParams.get('includeOutdated') === 'true';
+  const toggleIncludeOutdated = (): void => {
+    setParam('includeOutdated', filter.includeOutdated ? 'false' : 'true');
+  };
+
+  const studyList = props.studies
+    .get(currentSeed)
+    ?.studies(filter)
+    .map((study, i) => (
+      <StudyItem key={study.name() + i} study={study} filter={filter} />
+    ));
+
+  return (
+    <div className="row">
+      <div className="col-sm-12">
+        <div className="card mb-3">
+          <div className="card-header filter-list">
+            <FilterCheckbox
+              title="Show empty groups"
+              htmlName="showEmptyGroups"
+              checked={filter.showEmptyGroups}
+              toggle={toggleShowEmptyGroups}
+            />
+            <FilterCheckbox
+              title="Show outdated studies"
+              htmlName="includeOutdated"
+              checked={filter.includeOutdated}
+              toggle={toggleIncludeOutdated}
+            />
+          </div>
+        </div>
+        {studyList}
+      </div>
+    </div>
+  );
+}
+
+class AppState {
+  studies = new Map<SeedType, StudyListModel>();
+}
+
+export function App(): JSX.Element {
+  const [state, setState] = React.useState(new AppState());
+  React.useEffect(() => {
+    const load = async (url: string, type: SeedType): Promise<void> => {
+      const studyList = await loadSeed(url, type);
+      setState((prevState) => {
+        const newState: AppState = { ...prevState };
+        if (studyList !== undefined) newState.studies.set(type, studyList);
+        return newState;
+      });
+    };
+    load(core_utils.variationsProductionUrl, SeedType.PRODUCTION).catch(
+      console.error,
+    );
+    load(core_utils.variationsStagingUrl, SeedType.STAGING).catch(
+      console.error,
+    );
+    load(core_utils.variationsUpstreamUrl, SeedType.UPSTREAM).catch((e) => {});
+  }, []);
+
+  const hasUpstream = state.studies.get(SeedType.UPSTREAM) !== undefined;
+  const upstreamTab = hasUpstream ? (
+    <NavItem type={SeedType.UPSTREAM} />
+  ) : (
+    <></>
+  );
+  return (
+    <div className="container" id="app">
+      <section className="navbar navbar-light bg-light">
+        <h1>Brave Variations</h1>
+        <nav className="nav nav-pills">
+          <NavItem type={SeedType.PRODUCTION} />
+          <NavItem type={SeedType.STAGING} />
+          {upstreamTab}
```

Comment:
the conditional can be inline for readability
```tsx
{hasUpstream &&
<NavItem type={SeedType.UPSTREAM} />
}
```

Reply:
Done